### PR TITLE
Implementing light range support in Metal

### PR DESCRIPTION
**Pull request by colincornaby**
This fixes the popping issue discovered while investigating bump maps in BumpDemo.

Similar changes may need to be done to OpenGL. These changes will also need to be back ported into the per pixel lighting branch.

### CODE REVIEW

**Review by dpogue — APPROVED** [review]

File: Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp [hunk 1]
```diff
@@ -2374,6 +2374,8 @@ void plMetalPipeline::IEnableLight(size_t i, plLightInfo* light)
     plDirectionalLightInfo* dirLight = nullptr;
     plOmniLightInfo*        omniLight = nullptr;
     plSpotLightInfo*        spotLight = nullptr;
+    
+    const float             maxRange = 32767.f;
```
[the remark this answers is not in the document]
Comment (Hoikas) [reply Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp:2378]: Looks like this is not actually "outdated"

File: Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp [hunk 2]
```diff
@@ -2383,15 +2386,18 @@ void plMetalPipeline::IEnableLight(size_t i, plLightInfo* light)
         fLights.lampSources[i].constAtten = 1.0f;
         fLights.lampSources[i].linAtten = 0.0f;
         fLights.lampSources[i].quadAtten = 0.0f;
+
```
Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp:2389]:
```suggestion
```

Spurious whitespace change

File: Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (colincornaby) [reply Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp:2378]: I'm confused. Was there a comment somewhere labeling this as outdated?

[the remark this answers is not in the document]
Comment (Hoikas) [reply Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp:2378]: GitHub says this chain is outdated

**Review by Hoikas — APPROVED** [review]

**Pull request merged**